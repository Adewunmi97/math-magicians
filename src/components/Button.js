import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styles from '../assets/button.module.css';

// eslint-disable-next-line react/prefer-stateless-function
class Button extends Component {
  render() {
    const { name, handleClick } = this.props;
    let classes = styles.Button;
    if (name === '0') classes += ` ${styles.ButtonSpan2}`;
    if (name === 'x' || name === 'รท' || name === '-' || name === '+' || name === '=') classes += ` ${styles.ButtonOrange}`;

    const onClickHandler = () => handleClick(name);

    return (
      <button type="button" className={`${classes}`} onClick={onClickHandler}>
        {name}
      </button>
    );
  }
}

Button.propTypes = {
  name: PropTypes.string.isRequired,
  handleClick: PropTypes.func.isRequired,
};

export default Button;
