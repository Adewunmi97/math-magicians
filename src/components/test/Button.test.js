import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Button from '../Button';

describe('test Button Functionality', () => {
  it('should render Button correctly', () => {
    const { container } = render(<Button name="0" handleClick={() => {}} />);
    expect(screen.getByRole('button')).toHaveTextContent('0');
    expect(container).toMatchSnapshot();
  });
});
