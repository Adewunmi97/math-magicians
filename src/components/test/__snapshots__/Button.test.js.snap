// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Button Functionality should render Button correctly 1`] = `
<div>
  <button
    class="Button ButtonSpan2"
    type="button"
  >
    0
  </button>
</div>
`;
